models/object_detection_nanodet/NanodetPlus.py

```diff
@@ -0,0 +1,138 @@
+    def __init__(self, modelPath, prob_threshold=0.35, iou_threshold=0.6):
+        with open('coco.names', 'rt') as f:
+            self.classes = f.read().rstrip('\n').split('\n')
+
+        self.num_classes = len(self.classes)
+        self.strides = (8, 16, 32, 64)
+        self.image_shape = (416, 416)
+        self.reg_max = 7
+        self.prob_threshold = prob_threshold
+        self.iou_threshold = iou_threshold
+        self.project = np.arange(self.reg_max + 1)
+        self.mean = np.array([103.53, 116.28, 123.675], dtype=np.float32).reshape(1, 1, 3)
+        self.std = np.array([57.375, 57.12, 58.395], dtype=np.float32).reshape(1, 1, 3)
+        self.net = cv2.dnn.readNet(modelPath)
+
+        self.anchors_mlvl = []
+        for i in range(len(self.strides)):
+            featmap_size = (int(self.image_shape[0] / self.strides[i]), int(self.image_shape[1] / self.strides[i]))
+            stride = self.strides[i]
+            feat_h, feat_w = featmap_size
+            shift_x = np.arange(0, feat_w) * stride
+            shift_y = np.arange(0, feat_h) * stride
+            xv, yv = np.meshgrid(shift_x, shift_y)
+            xv = xv.flatten()
+            yv = yv.flatten()
+            cx = xv + 0.5 * (stride-1)
+            cy = yv + 0.5 * (stride - 1)
+            #anchors = np.stack((cx, cy), axis=-1)
+            anchors = np.column_stack((cx, cy))
+            self.anchors_mlvl.append(anchors)
+
+    def softmax_func(self,x, axis=0):
+        x_exp = np.exp(x)
+        x_sum = np.sum(x_exp, axis=axis, keepdims=True)
+        s = x_exp / x_sum
+        return s
+
+    def pre_process(self, img):
+        img = img.astype(np.float32)
+        img = (img - self.mean) / self.std
+        blob = cv2.dnn.blobFromImage(img)
+        return blob
+
+    def infer(self, srcimg, keep_ratio=True):
+        top, left, newh, neww = 0, 0, self.image_shape[0], self.image_shape[1]
+        if keep_ratio and srcimg.shape[0] != srcimg.shape[1]:
+            hw_scale = srcimg.shape[0] / srcimg.shape[1]
+            if hw_scale > 1:
+                newh, neww = self.image_shape[0], int(self.image_shape[1] / hw_scale)
+                img = cv2.resize(srcimg, (neww, newh), interpolation=cv2.INTER_AREA)
+                left = int((self.image_shape[1] - neww) * 0.5)
+                img = cv2.copyMakeBorder(img, 0, 0, left, self.image_shape[1] - neww - left, cv2.BORDER_CONSTANT,
+                                         value=0)  # add border
+            else:
+                newh, neww = int(self.image_shape[0] * hw_scale), self.image_shape[1]
+                img = cv2.resize(srcimg, (neww, newh), interpolation=cv2.INTER_AREA)
+                top = int((self.image_shape[0] - newh) * 0.5)
+                img = cv2.copyMakeBorder(img, top, self.image_shape[0] - newh - top, 0, 0, cv2.BORDER_CONSTANT, value=0)
+        else:
+            img = cv2.resize(srcimg, self.image_shape, interpolation=cv2.INTER_AREA)
+
+        blob = self.pre_process(img)
```

Comment:
Can we reuse the code from yolox for letterbox preprocessing?

Reply:
Need this type of pre-processing for nanodet, but now I have put these steps in the pre-process function similar to YoloX and the infer function has only the following steps
- preprocess
- infer
- postprocess